### ISSUE TITLE
Business themes: Missing editor palette in functions.php

### ISSUE THREAD

**Issue by Copons** [Automattic/themes#1179]
## Steps to replicate

- Activate a Business theme (e.g. Modern Business).
- Edit a post or page, insert a new paragraph and apply to it some custom colors (try with primary or secondary). Notice that in the palette the primary and secondary colors are blue.
- Preview the post. Notice that the paragraph appears pink instead.

## Result

| Editor | Front end |
| - | - |
| ![Screenshot 2019-08-02 at 13 20 07](https://user-images.githubusercontent.com/2070010/62369928-101af700-b529-11e9-9201-4fe8c774a688.png) | ![Screenshot 2019-08-02 at 13 20 47](https://user-images.githubusercontent.com/2070010/62369949-20cb6d00-b529-11e9-9002-4d922ae1e2d1.png) |

## Expected

| Editor | Front end |
| - | - |
| ![Screenshot 2019-08-02 at 13 21 16](https://user-images.githubusercontent.com/2070010/62369943-1c9f4f80-b529-11e9-9eb4-8c811af9ecea.png) | ![Screenshot 2019-08-02 at 13 20 47](https://user-images.githubusercontent.com/2070010/62369949-20cb6d00-b529-11e9-9002-4d922ae1e2d1.png) |

## Explanation and suggested fix

Editor color palettes require two parts to work:
- One in PHP where we define the colors names and codes, which is used by the editor.
- One in CSS where we assign styles to the classes automagically generated by the editor.

Business themes only have the CSS part, while inheriting the PHP one from Twenty Nineteen.
This means that, for example, the Primary color is blue in the editor (as defined by [Twenty Nineteen `functions.php`](https://github.com/WordPress/twentynineteen/blob/e74d348ab9cfc6ad46bdf33f691611930c39e944/functions.php#L141-L171)), and pink on the front-end (as defined by [Modern Business CSS](https://github.com/Automattic/themes/blob/0afe9f26f9b91d3710fa5a9d32ac5b1d9a8210ce/modern-business/sass/blocks/_blocks.scss#L1045-L1098)).

I've opened a PR with the fix for Modern Business: https://github.com/Automattic/themes/pull/1177
It should be enough to copy it on the other themes, and updating the colors by looking in the `_blocks.scss` and `_colors.scss` files.

**Comment by iamtakashi** [comment]
Fixed in #1177.

**Comment by iamtakashi** [comment]
Oh, sorry. I closed this too soon. We still need to fix other themes :)

**Comment by Copons** [comment]
@iamtakashi  Actually, there's something weird going on here. 😄 

My PR doesn't harm but, as you pointed out, it might not be needed.
I'm still figuring out what happens, so please don't waste time on this issue for now. 🙂

**Comment by Copons** [comment]
Ok, this is apparently quite odd.

I noticed the issue and tested my fix on my local WP install, which uses a WPORG install completely untied from WPCOM.
Most notably, the version of Twenty Nineteen I have is the 1.4 (May 2019).

As it turns out, all Business themes override the Twenty Nineteen palette by using some Twenty Nineteen filters like `twentynineteen_default_hue` that, by setting a base HSL, automagically generate the entire palette.
https://github.com/Automattic/themes/blob/daa9ae084e447c341602da7014f4c8bb68417e08/modern-business/inc/wpcom.php

The filters are defined in the `twentynineteen/inc/color-filters.php` file, which I can't find in the Twenty Nineteen repo because it doesn't exist anymore.
I believe it was removed between the Twenty Nineteen version 1.3 and 1.4, because:
- I have it in another, outdated, local install on Twenty Nineteen 1.3.
- Twenty Nineteen on WPCOM has got it, and apparently it's a custom `1.3-wpcom` version.

In other words, **those filters we use in the Business theme, work fine when inheriting an outdated Twenty Nineteen, but not at all on an updated one**.

My solution fixes the issue on WPORG sites, and doesn't harm WPCOM ones.
I think we could extend it to the other Business themes, and also delete the outdated color filters.
Though, I'd love to have the approval from theme folks, because I'm a bit rusty with all this stuff.

**Comment by iamtakashi** [comment]
Thanks for looking into it. I agree, adding the `editor-color-palette` support to other business themes seems a good idea. We'll do that. Thanks again!

**Comment by cathymcbride** [comment]
retired theme. closing
